### PR TITLE
Updates python to 3.12

### PR DESCRIPTION
**Pull request by smithcommajoseph**
Bumps all packages/deps
Removes pylama (no longer maintained)

### CODE REVIEW

**Comment by haneslinger** [comment]
>  Removes pylama (no longer maintained)

😢 do we want to add in the things pylama used to package? or a different code auditor?

**Comment by haneslinger** [comment]
> Removes pylama (no longer maintained)
Also,  I see it hasn't been touched in two years, but did it announce its wintering? did they just ghost? I can find anything

File: pyproject.toml [hunk 1]
```diff
@@ -9,32 +9,32 @@ authors = ["Your Name <you@example.com>"]
 repository = "https://github.com/NREL/Wattile"
```
Comment (haneslinger) [inline pyproject.toml:9]: noice

**Comment by smithcommajoseph** [comment]
Looks like pylama has been ghosted. There's even an outstanding PR that fixes the python 3.12 issues [here](https://github.com/klen/pylama/pull/254), but no love from the maintainer.

As for pylama replacement, I don't fully understand all the things it did, so I'm ill equipped to recommend a replacement. Do _you_ have preferences? I dig into this in the coming days

**Review by haneslinger — APPROVED** [review]
Looking good, thank you.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,17 +1,11 @@
...
+    rev: 5.13.2
     hooks:
     -    id: isort
-         entry: isort --check
--   repo: https://github.com/gvanderest/pylama-pre-commit
-    rev: 0.1.2
-    hooks:
-    -   id: pylama
-        additional_dependencies:
-        -   pyflakes==2.4.0
```
[the remark this answers is not in the document]
Comment (smithcommajoseph) [reply .pre-commit-config.yaml:17]: pulled the pylama pre-commit due to the pylama errors

**Pull request merged**